camel/storages/vectordb_storages/milvus.py

```diff
@@ -0,0 +1,360 @@
+                raise ValueError(
+                    "Vector dimension of the existing collection "
+                    f'"{self.collection_name}" ({in_dim}) is different from '
+                    f"the given embedding dim ({self.vector_dim}).")
+        else:
+            self._create_collection(collection_name=self.collection_name, )
+
+    def _create_collection(
+        self,
+        collection_name: str,
+        **kwargs: Any,
+    ) -> None:
+        r"""Creates a new collection in the database.
+
+        Args:
+            collection_name (str): Name of the collection to be created.
+            **kwargs (Any): Additional keyword arguments.
+        """
+        from pymilvus import CollectionSchema, DataType, FieldSchema
+
+        index_params = {
+            "metric_type": "COSINE",
+            "index_type": "AUTOINDEX",
+            "params": {},
+        }
+
+        fields = [
+            FieldSchema(name='id', dtype=DataType.VARCHAR,
+                        descrition='A unique identifier for the vector',
+                        is_primary=True, auto_id=False, max_length=65535),
+            FieldSchema(
+                name='payload', dtype=DataType.VARCHAR,
+                description=('Any additional metadata or information related'
+                             'to the vector'), max_length=65535),
+            FieldSchema(
+                name='vector', dtype=DataType.FLOAT_VECTOR,
+                description='The numerical representation of the vector',
+                dim=self.vector_dim)
+        ]
+        schema = CollectionSchema(fields=fields,
+                                  description='Collection schema')
+
+        self._client.create_collection_with_schema(
+            collection_name=collection_name,
+            schema=schema,
+            index_params=index_params,
+            **kwargs,
+        )
+
+    def _delete_collection(
+        self,
+        collection_name: str,
+    ) -> None:
+        r"""Deletes an existing collection from the database.
+
+        Args:
+            collection (str): Name of the collection to be deleted.
+        """
+        self._client.drop_collection(collection_name=collection_name)
+
+    def _collection_exists(self, collection_name: str) -> bool:
+        r"""Checks whether a collection with the specified name exists in the
+        database.
+
+        Args:
+            collection_name (str): The name of the collection to check.
+
+        Returns:
+            bool: True if the collection exists, False otherwise.
+        """
+        for c in self._client.list_collections():
+            if collection_name == c:
+                return True
+        return False
+
+    def _generate_collection_name(self) -> str:
+        r"""Generates a unique name for a new collection based on the current
+        timestamp.
+
+        Returns:
+            str: A unique collection name.
+        """
+        return datetime.now().isoformat()
```

Comment:
This will get invalid collection name. Check milvus collection name rules: "Name contains only alphanumeric letters and underscores."

Reply:
thanks for pointing this out, have added validation for generated collection name